### PR TITLE
Request context

### PR DESCRIPTION
**Pull request by TMSCH**
Allows to pass a `context` param to `retrieve` and `hint` request. It can contain one of the three different context for a request:

- `'signUp'`: for account creation
- `'signIn'`: for signing in an user
- `'continue'`: for other use cases

It's passed through to the provider's implementation.

Also allows to use the CredentialManagement in localhost.

### CODE REVIEW

File: ts/protocol/utils.ts [hunk 1]
```diff
@@ -132,7 +132,9 @@ export function stringValidator(value: any): boolean {
  * Returns true if the current origin is https.
  */
 export function isSecureOrigin(): boolean {
-  return window.location.protocol === 'https:';
+  return window.location.protocol === 'https:' ||
```
Comment (iainmcgin) [inline ts/protocol/utils.ts]: This change seems unrelated?

Comment (TMSCH) [inline ts/protocol/utils.ts]: It is unrelated to the Request Context indeed, but I mention it in the PR description. The PR was pretty small so I wanted to make that change here, is that OK?

Comment (iainmcgin) [inline ts/protocol/utils.ts]: Yeah that's fine, you can leave it here for expediency. I missed it in the PR description. Generally I prefer isolating unrelated changes.

Comment (TMSCH) [inline ts/protocol/utils.ts]: Thanks. I totally agree, but it takes time to get these PRs in and incorporate them in our code, so for such a small change I assumed it'd be fine.
